src/deepsparse/evaluation/evaluator.py

```diff
@@ -0,0 +1,226 @@
+    )
+
+    parser.add_argument(
+        "-s",
+        "--splits",
+        type=Optional[List[str], str],
+        default=None,
+        help="The name of the splits to evaluate on. "
+        "Can be a string for a single split "
+        "or a list of strings for multiple splits.",
+    )
+
+    parser.add_argument(
+        "-m",
+        "--metrics",
+        type=Optional[List[str], str],
+        default=None,
+        help="The name of the metrics to evaluate on. "
+        "Can be a string for a single metric "
+        "or a list of strings for multiple metrics.",
+    )
+    parser.add_argument(
+        "--enforce-result-structure",
+        "--enforce_result_structure",
+        action="store_true",
+        help="Specifies whether to unify all the results "
+        "into the predefined Evaluation structure. "
+        "If True, the results will be returned as a "
+        "list of Evaluation objects. Otherwise, the "
+        "result will preserve the original result "
+        "structure from the evaluation integration.",
+    )
+
+
+def evaluate(
+    target: str,
+    datasets: Union[str, List[str]],
+    integration: str,
+    engine_type: Union[
+        DEEPSPARSE_ENGINE, ORT_ENGINE, TORCHSCRIPT_ENGINE, None
+    ] = DEEPSPARSE_ENGINE,
+    batch_size: int = 1,
+    splits: Union[List[str], str, None] = None,
+    metrics: Union[List[str], str, None] = None,
+    enforce_result_structure: bool = True,
+    **kwargs,
+) -> Union[List[Evaluation], Any]:
+
+    _LOGGER.info(f"Target to evaluate: {target}")
+    if engine_type:
+        _LOGGER.info(f"A pipeline with the engine type: {engine_type} will be created")
+    else:
+        _LOGGER.info(
+            "No engine type specified. The target "
+            "will be evaluated using the default framework"
+        )
+    _LOGGER.info(f"Datasets to evaluate on: {datasets}")
+    _LOGGER.info(
+        f"Batch size: {batch_size}\n"
+        f"Splits to evaluate on: {splits}\n"
+        f"Metrics to evaluate on: {metrics}"
+    )
+
+    eval_integration = EvaluationRegistry.load_from_registry(integration)
+
+    _LOGGER.info(
+        f"The following evaluation integration has "
+        f"been successfully setup: {eval_integration.__name__}"
+    )
+
+    result = eval_integration(
+        target=target,
+        datasets=datasets,
+        engine_type=engine_type,
+        batch_size=batch_size,
+        splits=splits,
+        metrics=metrics,
+        original_result_structure=enforce_result_structure,
+        **kwargs,
+    )
+
+    if enforce_result_structure:
+        if not validate_result_structure(result):
+            raise ValueError(
+                "The evaluation integration must return a list of Evaluation objects "
+                "when enforce_result_structure is True."
+            )
+        _LOGGER.info(f"Evaluation done. Results:\n{print_result(result)}")
+
+    return result
+
+
+def main():
```

Comment:
API should not print, the CLI should. On the start and at the end. Option to save through the CLI